AutoSteamApp/Automaton/SteamworkAutomaton.cs

```diff
@@ -0,0 +1,348 @@
+            // If the match is made
+            if (match.Success)
+                // And we have a capture group
+                if (match.Groups.Count > 1)
+                    // Try to turn it into a number
+                    if (int.TryParse(match.Groups[1].Value, out int result))
+                    {
+                        Log.Message("Version found: " + result);
+                        // If it is a numeber and is the same as the supported version
+                        if (result == MHWMemoryValues.SupportedGameVersion)
+                        {
+                            // Set the flag
+                            _SupportedVersion = true;
+                            return;
+                        }
+                        Log.Warning(
+                            "Version unsupported. Currently supported version: " + MHWMemoryValues.SupportedGameVersion +
+                            "\n\t\tAutomaton will still run, however, correct sequences cannot be read"
+                                    );
+                        return;
+                    }
+            Log.Error(
+                "Could not verify game version. This is most likely due to a different versioning system being used by Capcom." +
+                "\n\t\tAutomaton will still run, however, correct sequences cannot be read"
+                     );
+        }
+
+        /// <summary>
+        /// Extracts from the process the correct sequence to input, then inputs it.
+        /// </summary>
+        /// <param name="cts">Cancellation token used to signal when to stop.</param>
+        void ExtractAndEnterSequence(CancellationToken cts)
+        {
+            try
+            {
+                // Generate a sequence to input
+                VirtualKeyCode[] sequence;
+
+                // If the version is supported, use the extracted sequence
+                sequence = _SteamworksData.ExtractSequence();
+
+                if (sequence == null)
+                {
+                    Log.Debug("Could not find a valid sequence. Are you sure you are in the game?");
+                    Thread.Sleep(1000);
+                    return;
+                }
+
+                //Here we need to check the probability of us winning based on the rarity of the reward
+                float probability = ConfigurationReader.CommonSuccessRate;
+                if (_SteamworksData.RewardRarityValue == RewardRarity.Rare)
+                {
+                    Log.Debug("Rare reward detected.");
+                    probability = ConfigurationReader.RareSuccessRate;
+                }
+
+                // Use rng to check if we win or not
+                // TODO: maybe not create a new instance every time? I'll need to consult with someone about the probability distribution
+                // when doing it this way.
+                Random rng = new Random();
```

Comment:
You should use the same random object to ensure a correct distribution. You can see here (https://docs.microsoft.com/en-us/dotnet/api/system.random?redirectedfrom=MSDN&view=netcore-3.1) that using a "new" object you could get the same values in the same order - if nothing else changes. It is a good practice to have a Random object which is reused every time you need a new "random" value.